ecommerce/CartController.php

```diff
@@ -121,64 +136,58 @@ public static function get_page_id() {
 	}
 
 	/**
-	 * Retrieve cart data by cart ID and respond with JSON.
+	 * Get cart items
 	 *
-	 * This function retrieves the cart ID from the POST request, validates it,
-	 * fetches the corresponding cart data using the CartModel class, and returns
-	 * a JSON response with the cart data or an error message.
+	 * @since 3.0.0
 	 *
-	 * If the cart ID is not provided, it responds with a "Bad Request" status.
-	 * If the cart is not found, it responds with a "Not Found" status.
-	 * Otherwise, it responds with the cart data and a success message.
+	 * @return array
+	 */
+	public function get_cart_items() {
+		$user_id = tutils()->get_user_id();
+		return $this->model->get_cart_items( $user_id );
+	}
+
+	/**
+	 * Add course to cart
 	 *
 	 * @since 3.0.0
 	 *
-	 * @return void
+	 * @return void JSON response
 	 */
-	public function get_cart_by_id() {
+	public function tutor_add_course_to_cart() {
 		if ( ! tutor_utils()->is_nonce_verified() ) {
-			$this->json_response( tutor_utils()->error_message( 'nonce' ), null, HttpHelper::STATUS_BAD_REQUEST );
+			wp_send_json_error( tutor_utils()->error_message( 'nonce' ) );
 		}
 
-		$cart_id = Input::post( 'cart_id' );
+		$user_id   = tutils()->get_user_id();
+		$course_id = Input::post( 'course_id' );
 
-		if ( empty( $cart_id ) ) {
-			$this->json_response(
-				__( 'Cart ID is required', 'tutor' ),
-				null,
-				HttpHelper::STATUS_BAD_REQUEST
-			);
+		if ( ! is_numeric( $course_id ) ) {
+			wp_send_json_error( __( 'Invalid course id.', 'tutor' ) );
 		}
 
-		$cart_data = $this->model->get_cart_by_id( $cart_id );
-
-		if ( ! $cart_data ) {
-			$this->json_response(
-				__( 'Cart not found', 'tutor' ),
-				null,
-				HttpHelper::STATUS_NOT_FOUND
-			);
+		// Check if the course already exists in the cart or not.
+		$is_course_in_user_cart = $this->model->is_course_in_user_cart( $user_id, $course_id );
+		if ( $is_course_in_user_cart ) {
+			wp_send_json_error( __( 'The course is already in the cart.', 'tutor' ) );
 		}
 
-		$this->json_response(
-			__( 'Cart retrieved successfully', 'tutor' ),
-			$cart_data
-		);
-	}
+		$response = $this->model->add_course_to_cart( $user_id, $course_id );
 
-	/**
-	 * Get cart items
-	 *
-	 * @since 3.0.0
-	 *
-	 * @return array
-	 */
-	public function get_cart_items() {
-		return $this->model->get_cart_items();
+		if ( $response ) {
+			wp_send_json_success(
+				array(
+					'message'       => __( 'The course was added to the cart successfully.', 'tutor' ),
```

Comment:
Please make the response consistent just directly pass cart page URL in the body